### DISCUSSION TITLE
vscode-test throw unexpected error `Fail`

### DISCUSSION THREAD

**Discussion by fi3ework** [community/community#26503] [Actions]
<p>workflow file: <a href="https://github.com/fi3ework/vscode-antd-rush/blob/master/.github/workflows/test.yml" rel="nofollow noopener">https://github.com/fi3ework/vscode-antd-rush/blob/master/.github/workflows/test.yml</a></p>
<p>a failed action: <a href="https://github.com/fi3ework/vscode-antd-rush/runs/599819681?check_suite_focus=true" rel="nofollow noopener">https://github.com/fi3ework/vscode-antd-rush/runs/599819681?check_suite_focus=true</a></p>
<p>I use vscode-test to test my VSCode extension. But recently it directly throws <code>Fail</code> when initializing. The error is throw from here <a href="https://github.com/microsoft/vscode-test/blob/master/lib/runTest.ts#L134" rel="nofollow noopener">https://github.com/microsoft/vscode-test/blob/master/lib/runTest.ts#L134</a>, but I don’t know why it throw. I can pass the test on my Macbook Pro(masOS 10.14.6, nodejs v10.16.0). I can’t  reproduce it on my own computer no matter how. :man_facepalming:</p>
<p>Please help me or give a hint where I might missed or wrong. Thank you very much. :pray:t2:</p>
<p>Here’s the failed screenshot :point_down:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/660fd7f48893d674746151fe08a89622c1f5b2fa.png" data-download-href="/uploads/short-url/eySE5S2tGB2WplzlgStKCvawbHA.png?dl=1" title="image.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12322740/181098078-5691a862-c7a7-43fb-b00a-c6162fa89edc.png" alt="image.png" data-base62-sha1="eySE5S2tGB2WplzlgStKCvawbHA" width="690" height="290" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/6/660fd7f48893d674746151fe08a89622c1f5b2fa_2_690x290.png, https://aws1.discourse-cdn.com/github/optimized/2X/6/660fd7f48893d674746151fe08a89622c1f5b2fa_2_1035x435.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/6/660fd7f48893d674746151fe08a89622c1f5b2fa_2_1380x580.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/660fd7f48893d674746151fe08a89622c1f5b2fa_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image.png</span><span class="informations">2272×958 245 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>and workflow file (pasted for better preview)</p>
<pre><code>name: Test

on: [push]

jobs:
  build:
    runs-on: ${{ matrix.os }}
    strategy:
      fail-fast: false
      matrix:
        os: [macos-latest, windows-latest]
        node: [10, 12]
    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v1
        with:
          node-version: ${{ matrix.node-version }}
      - name: yarn - install &amp; compile
        run: |
          yarn
          yarn run compile
          yarn run compile-test
      - name: Run headless tests
        uses: GabrielBB/xvfb-action@v1.0
        with:
          run: npm run test</code></pre>

**Comment by fi3ework** [comment, marked as answer]
<p>Oh, I can reproduce it now.  VSCode on my computer has installed the extension(online version). It seems like covered the error from the unpacked extension. It should be a bug of my code :sweat_smile:.</p>

**Comment by heaths** [comment]
<p>Glad you solved it. Just a quick note, though: if you test under linux, you need to configure Xvfb and start it. With ubuntu-16.04, it’s pretty easy. You just need a step like so:</p>
<pre><code>- name: Start Xvfb
      run: |
        set -e
        /usr/bin/Xvfb :10 -ac &gt;&gt; /tmp/Xvfb.out 2&gt;&amp;1 &amp;
        disown -ar
      if: runner.os == 'Linux'
</code></pre>
<p>For ubuntu-18.04 / ubuntu-latest, see vscode’s own tests: <a href="https://github.com/microsoft/vscode/blob/df448dd078d60103a05fa8f068318abe443347d3/.github/workflows/ci.yml#L128-L135" rel="nofollow noopener">https://github.com/microsoft/vscode/blob/df448dd078d60103a05fa8f068318abe443347d3/.github/workflows/ci.yml#L128-L135</a></p>
